code/modules/security/genpop.dm

```diff
@@ -0,0 +1,650 @@
+	paperwork.add_raw_text("<h1 id='record-of-incarceration'>Record Of Incarceration:</h1> <hr> <h2 id='name'>Name: </h2> <p>[desired_name]</p> <h2 id='crime'>Crime: </h2> <p>[desired_crime]</p> <h2 id='sentence-min'>Sentence (Min)</h2> <p>[desired_sentence/60]</p> <h2 id='description'>Description </h2> <p>[desired_details]</p> <p>WhiteRapids Military Council, disciplinary authority</p>")
+	paperwork.update_appearance()
+	desired_name = null
+	desired_details = null
+	playsound(src, 'sound/machines/terminal_insert_disc.ogg', 50, 0)
+	next_print = world.time + 5 SECONDS
+
+
+
+
+/obj/machinery/genpop_interface/ui_act(action, params)
+	if(buildstage != 2 & panel_open)
+		return
+	if(isliving(usr))
+		playsound(src, 'sound/machines/terminal_prompt_confirm.ogg', 50, FALSE)
+	if(..())
+		return
+	if(!allowed(usr))
+		to_chat(usr, "<span class='warning'>Access denied.</span>")
+		return
+	switch(action)
+		if("prisoner_name")
+			// Encode the name and ensure it is saniticed for IC input
+			var/prisoner_name = stripped_input(usr, "Input prisoner's name...", "Crimes", desired_name)
+			prisoner_name = sanitize_name(prisoner_name)
+			if(!prisoner_name || (!Adjacent(usr) && !IsAdminGhost(usr)))
+				return FALSE
+			desired_name = prisoner_name
+			var/prisoner_details = stripped_input(usr, "Input details of the offense...", "Crimes", desired_details)
+			if (!prisoner_details || CHAT_FILTER_CHECK(prisoner_details) || (!Adjacent(usr) && !IsAdminGhost(usr)))
+				return TRUE
+			desired_details = prisoner_details
+			// Ask them for the details of the crime
+		if("edit_details")
+			var/prisoner_details = stripped_input(usr, "Input details of the offense...", "Crimes", desired_details)
+			if (!prisoner_details || CHAT_FILTER_CHECK(prisoner_details) || (!Adjacent(usr) && !IsAdminGhost(usr)))
+				return FALSE
+			desired_details = prisoner_details
+			// Ask them for the details of the crime
+		if("print")
+			if (!desired_name)
+				return
+			if (!desired_details)
+				return
+			var/desired_sentence = text2num(params["desired_sentence"])
+			if (!desired_sentence)
+				return
+			var/desired_crime = params["desired_crime"]
+			if (!valid_crime_name_regex.Find(desired_crime))
+				log_href_exploit(usr, "Entered a desired crime which was not permitted by the desired crime regex.")
+				return
+			print_id(usr, desired_crime, desired_sentence)
+
+		// For adjusting the time of pre-existing prisoners
+		if("adjust_time")
+			var/obj/item/card/id/prisoner/id = locate(params["id"])
+			var/value = text2num(params["adjust"])
+			if(!istype(id) || id.access == ACCESS_PRISONER) //check for prisonner access too
+				return
+			if(id.served_time >= id.sentence)
+				say("Prisoner has already served their time! Please apply another charge to sentence them with!")
+				return
+			if(value && isnum(value))
+				id.sentence += value
+				id.sentence = clamp(id.sentence,0,MAX_TIMER)
+
+		if("release")
+			var/obj/item/card/id/prisoner/id = locate(params["id"])
```

Comment:
Same here with limiting the scope of what can be located to the things inside the global prisoner ID list